front/app/components/MultiCrumb/MultiCrumb.tsx

```diff
@@ -2,145 +2,38 @@ import * as React from 'react';
-              const label = this.props.labels ? this.props.labels[i] : v;
-
-              return (
-                <b key={v}>
-                  <span className="crumb-container">
-                    {label}
-                    <FontAwesome
-                      className="remove crumb-icon"
-                      name="remove"
-                      onClick={() => this.props.onClick(v)}
-                    />
-                  </span>
-                </b>
-              );
-            })}
-            <b>
-              <span className="crumb-container">
-                {`...${addVals} others`}
-                <FontAwesome
-                  className="chevron-right crumb-icon"
-                  name="chevron-right"
-                  onClick={() => this.toggleShowValue()}
-                />
-              </span>
-            </b>
-          </ListGroupItem>
-        </MultiCrumb>
-      );
-    } else if (this.props.values.length > 4 && this.state.showValue == true) {
-      return (
-        <MultiCrumb>
-          <ListGroupItem className="filter-values">
-            {this.props.category && <i>{this.props.category}:</i>}
-            {this.props.values.map((v, i) => {
-              const label = this.props.labels ? this.props.labels[i] : v;
-              return (
-                <b key={v}>
-                  <span className="crumb-container">
-                    {label}
-                    <FontAwesome
-                      className="remove crumb-icon"
-                      name="remove"
-                      onClick={() => this.props.onClick(v)}
-                    />
-                  </span>
-                </b>
-              );
-            })}
+    const { category, values, agg, onClick, labels } = this.props;
 
-            <b>
-              <span className="shorten-crumb">
-                <FontAwesome
-                  className="chevron-left crumb-icon"
-                  name="chevron-left"
-                  onClick={() => this.toggleShowValue()}
-                />
-              </span>
-            </b>
-          </ListGroupItem>
-        </MultiCrumb>
-      );
-    } else {
-      return (
-        <MultiCrumb>
-          <ListGroupItem className="filter-values">
-            {this.props.category && <i>{this.props.category}:</i>}
-            {this.props.values.map((v, i) => {
-              const label = this.props.labels ? this.props.labels[i] : v;
-              return (
-                <b key={v}>
-                  <span className="crumb-container">
-                    {label}
-                    <FontAwesome
-                      className="remove crumb-icon"
-                      name="remove"
-                      onClick={() => this.props.onClick(v)}
-                    />
-                  </span>
-                </b>
-              );
-            })}
-          </ListGroupItem>
-        </MultiCrumb>
-      );
+    if (agg) {
```

Comment:
This lets us get slightly better separation of concerns from other places where multicrumb is used. The agg crumb knows how to display ranged values as well as multiple distinct scalar values.